data/src/main/java/com/microsoft/azure/kusto/data/ClientRequestProperties.java

```diff
@@ -0,0 +1,55 @@
+package com.microsoft.azure.kusto.data;
+
+import org.json.JSONObject;
+
+import java.util.HashMap;
+
+/*
+ * Kusto supports attaching various properties to client requests (such as queries and control commands).
+ * Such properties may be used to provide additional information to Kusto (for example, for the purpose of correlating client/service interaction),
+ * may affect what limits and policies get applied to the request, and much more.
+ * For a complete list of available client request properties
+ * check out https://docs.microsoft.com/en-us/azure/kusto/api/netfx/request-properties#list-of-clientrequestproperties
+ */
+public class ClientRequestProperties {
```

Comment:
There're unchecked conversions from Object to HashMap. I know we have that elsewhere, but this is a new class and probably there's a better way to deal with it...
One thing I don't understand about the usage of the HashMap - you have specific key strings defined as finals (inside the options key), meaning that you can do something similar with a well-defined class for the options. can't you?
Something like:
Class ClientRequestPropertiesOptions {
 private int serverTimeoutInMilliSecs;
 private int someOtherOptionKey;
 getters/setters
}

Reply:
not sure I understand what you mean..
I only explicitly provided 1 valid option, but there are others many other options.
I want to allow a somewhat dynamic way to manipulate those options, without making code cumbersome.

If you feel there is a better way to do so, please provide a code sample as I couldn't quite get what you meant above 👼

Reply:
I missed that there're methods that deal with an option with its name as an input.
I suggest to simplify the handling of the hashmaps with something like this:
`
public class ClientRequestProperties {
    private static final String OPTIONS_KEY = "Options";
    private static final String OPTION_SERVER_TIMEOUT = "servertimeout";

    private HashMap<String, Object> properties;
    private HashMap<String, Object> options;

    public ClientRequestProperties() {
        properties = new HashMap<>();
        options = new HashMap<>();
        properties.put(OPTIONS_KEY, options);
    }

    public void setOption(String name, Object value) {
        options.put(name, value);
    }

    public Object getOption(String name) {
        return options.get(name);
    }

    public void removeOption(String name) {
        options.remove(name);
    }

    public void clearOptions() {
        options.clear();
    }

    public Long getTimeoutInMilliSec() {
        return (Long) getOption(OPTION_SERVER_TIMEOUT);
    }

    public void setTimeoutInMilliSec(Long timeoutInMs) {
        setOption(OPTION_SERVER_TIMEOUT, timeoutInMs);
    }

    public JSONObject toJson() {
        return new JSONObject(properties);
    }

    public String toString() {
        return toJson().toString();
    }
}
`